smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/HttpProtocolTestGenerator.kt

```diff
@@ -42,16 +45,39 @@ class HttpProtocolTestGenerator(
     fun generateProtocolTests(): Int {
         val topDownIndex: TopDownIndex = TopDownIndex.of(ctx.model)
         val serviceSymbol = ctx.symbolProvider.toSymbol(ctx.service)
+        val operationMiddleware = updateRequestTestMiddleware()
         var numTests = 0
         for (operation in TreeSet(topDownIndex.getContainedOperations(ctx.service).filterNot(::serverOnly))) {
-            numTests += renderRequestTests(operation, serviceSymbol)
+            numTests += renderRequestTests(operation, serviceSymbol, operationMiddleware)
             numTests += renderResponseTests(operation, serviceSymbol)
             numTests += renderErrorTestCases(operation, serviceSymbol)
         }
         return numTests
     }
 
-    private fun renderRequestTests(operation: OperationShape, serviceSymbol: Symbol): Int {
+    private fun updateRequestTestMiddleware(): OperationMiddleware {
+        val topDownIndex: TopDownIndex = TopDownIndex.of(ctx.model)
+        val requestTestOperations = TreeSet(
+            topDownIndex.getContainedOperations(ctx.service)
+                .filter { it.hasTrait<HttpRequestTestsTrait>() }
+                .filterNot(::serverOnly)
+        )
+        val cloned = operationMiddleware.clone()
+
+        for (operation in requestTestOperations) {
+            cloned.removeMiddleware(operation, MiddlewareStep.BUILDSTEP, "EndpointResolverMiddleware")
+            cloned.removeMiddleware(operation, MiddlewareStep.BUILDSTEP, "UserAgentMiddleware")
+            cloned.removeMiddleware(operation, MiddlewareStep.FINALIZESTEP, "RetryMiddleware")
+            cloned.removeMiddleware(operation, MiddlewareStep.FINALIZESTEP, "AWSSigningMiddleware") // causes tests to halt :(
+            cloned.removeMiddleware(operation, MiddlewareStep.DESERIALIZESTEP, "DeserializeMiddleware")
+            cloned.removeMiddleware(operation, MiddlewareStep.DESERIALIZESTEP, "LoggingMiddleware")
+
+            cloned.appendMiddleware(operation, RequestTestEndpointResolverMiddleware(ctx.model, ctx.symbolProvider))
```

Comment:
(not in this PR.. but)Now that we are cloning this, we can create a new middleware to replace the function `renderMockDeserializeMiddleware`